### PR TITLE
Updated EthtoolBSD get_device_enabled_features

### PR DESCRIPTION
**Pull request by r-dailey**
Updated the EthtoolBSD get_device_enabled_features to correctly handle the formatting of the features list. Also updated the comment describing the formatting of the list.

### CODE REVIEW

**Comment by LiliDeng** [comment]
please run test by yourself.

I met below exception when run against image thefreebsdfoundation freebsd-14_0 14_0-release 14.0.0

```
2024-08-15T12:15:56.3051903Z 2024-08-15 12:15:56.300[5172][ERROR] lisa.suite[NetworkSettings].case[verify_device_enabled_features] case failed
2024-08-15T12:15:56.3052538Z Traceback (most recent call last):
2024-08-15T12:15:56.3053151Z   File "D:\a\_work\1\s\lisa\lisa\testsuite.py", line 780, in __run_case
2024-08-15T12:15:56.3053634Z     _call_with_timeout(
2024-08-15T12:15:56.3054109Z   File "D:\a\_work\1\s\lisa\lisa\testsuite.py", line 61, in _call_with_timeout
2024-08-15T12:15:56.3054656Z     func_timeout(
2024-08-15T12:15:56.3055213Z   File "C:\hostedtoolcache\windows\Python\3.10.16\x64\lib\site-packages\func_timeout\dafunc.py", line 108, in func_timeout
2024-08-15T12:15:56.3055839Z     raise_exception(exception)
2024-08-15T12:15:56.3057782Z   File "C:\hostedtoolcache\windows\Python\3.10.16\x64\lib\site-packages\func_timeout\py3_raise.py", line 7, in raise_exception
2024-08-15T12:15:56.3058511Z     raise exception[0] from None
2024-08-15T12:15:56.3058950Z   File "D:\a\_work\1\s\lisa\lisa\testsuite.py", line 505, in wrapper
2024-08-15T12:15:56.3059391Z     func(*args, **parameters)
2024-08-15T12:15:56.3059930Z   File "D:\a\_work\1\s\lisa\microsoft\testsuites\network\networksettings.py", line 267, in verify_device_enabled_features
2024-08-15T12:15:56.3060510Z     devices_features = ethtool.get_all_device_enabled_features()
2024-08-15T12:15:56.3061014Z   File "D:\a\_work\1\s\lisa\lisa\tools\ethtool.py", line 966, in get_all_device_enabled_features
2024-08-15T12:15:56.3061569Z     self.get_device_enabled_features(device, force_run)
2024-08-15T12:15:56.3062073Z   File "D:\a\_work\1\s\lisa\lisa\tools\ethtool.py", line 1120, in get_device_enabled_features
2024-08-15T12:15:56.3062590Z     features.append(self._get_bsd_to_linux_features_map[feature])
2024-08-15T12:15:56.3063052Z KeyError: 'JUMBO_MTU'
```

File: lisa/tools/ethtool.py [hunk 1]
```diff
@@ -1108,11 +1108,12 @@ def get_device_enabled_features(
         interface_info = self.node.tools[Ip].run(interface).stdout
 
         # Example output:
-        # options=8051b<RXCSUM,TXCSUM,VLAN_MTU,VLAN_HWTAGGING,TSO4,LRO,LINKSTATE>
-        # The features are separated by comma and enclosed by "<>"
+        # features=((RXCSUM,TXCSUM,VLAN_MTU,VLAN_HWTAGGING,TSO4,LRO,LINKSTATE), "/t")
```
Comment (squirrelsc) [inline lisa/tools/ethtool.py]: It's better to update the pattern `_interface_features_pattern`, or add a new pattern to handle both cases.

Comment (squirrelsc) [reply lisa/tools/ethtool.py]: You can use some online tool debugging regexp, like https://regex101.com/.

Comment (r-dailey) [reply lisa/tools/ethtool.py]: Oh I misunderstood this comment and the regex filter is still correct but the find group function is returning a tuple that looks like what I wrote. I will revert the comment and then check the other images for missing feature translations so there aren't any issues like Lili noted.

Comment (squirrelsc) [reply lisa/tools/ethtool.py]: You can update the pattern, so it can deal with the extra tuple.

**Comment by r-dailey** [comment]
> please run test by yourself.
> 
> I met below exception when run against image thefreebsdfoundation freebsd-14_0 14_0-release 14.0.0
> 
> ```
> 2024-08-15T12:15:56.3051903Z 2024-08-15 12:15:56.300[5172][ERROR] lisa.suite[NetworkSettings].case[verify_device_enabled_features] case failed
> 2024-08-15T12:15:56.3052538Z Traceback (most recent call last):
> 2024-08-15T12:15:56.3053151Z   File "D:\a\_work\1\s\lisa\lisa\testsuite.py", line 780, in __run_case
> 2024-08-15T12:15:56.3053634Z     _call_with_timeout(
> 2024-08-15T12:15:56.3054109Z   File "D:\a\_work\1\s\lisa\lisa\testsuite.py", line 61, in _call_with_timeout
> 2024-08-15T12:15:56.3054656Z     func_timeout(
> 2024-08-15T12:15:56.3055213Z   File "C:\hostedtoolcache\windows\Python\3.10.16\x64\lib\site-packages\func_timeout\dafunc.py", line 108, in func_timeout
> 2024-08-15T12:15:56.3055839Z     raise_exception(exception)
> 2024-08-15T12:15:56.3057782Z   File "C:\hostedtoolcache\windows\Python\3.10.16\x64\lib\site-packages\func_timeout\py3_raise.py", line 7, in raise_exception
> 2024-08-15T12:15:56.3058511Z     raise exception[0] from None
> 2024-08-15T12:15:56.3058950Z   File "D:\a\_work\1\s\lisa\lisa\testsuite.py", line 505, in wrapper
> 2024-08-15T12:15:56.3059391Z     func(*args, **parameters)
> 2024-08-15T12:15:56.3059930Z   File "D:\a\_work\1\s\lisa\microsoft\testsuites\network\networksettings.py", line 267, in verify_device_enabled_features
> 2024-08-15T12:15:56.3060510Z     devices_features = ethtool.get_all_device_enabled_features()
> 2024-08-15T12:15:56.3061014Z   File "D:\a\_work\1\s\lisa\lisa\tools\ethtool.py", line 966, in get_all_device_enabled_features
> 2024-08-15T12:15:56.3061569Z     self.get_device_enabled_features(device, force_run)
> 2024-08-15T12:15:56.3062073Z   File "D:\a\_work\1\s\lisa\lisa\tools\ethtool.py", line 1120, in get_device_enabled_features
> 2024-08-15T12:15:56.3062590Z     features.append(self._get_bsd_to_linux_features_map[feature])
> 2024-08-15T12:15:56.3063052Z KeyError: 'JUMBO_MTU'
> ```

This is now resolved but the Feature mapping from FreeBSD to Linux has several empty mappings as I couldn't easily identify them. I reached out to Wei to see if he knows more but any additionally knowledge would be great.

File: lisa/tools/ethtool.py (hunk 1 above)
Comment (r-dailey) [reply lisa/tools/ethtool.py]: I will do that.

**Comment by squirrelsc** [comment]
> This is now resolved but the Feature mapping from FreeBSD to Linux has several empty mappings as I couldn't easily identify them. I reached out to Wei to see if he knows more but any additionally knowledge would be great.

You don't need to list all of them. If there is no mapping, just ignore them.

**Review by LiliDeng — DISMISSED** [review]

**Comment by squirrelsc** [comment]
@r-dailey please monitor and fix checks.

**Comment by r-dailey** [comment]
> @r-dailey please monitor and fix checks.

Was already working on a fix should be resolved now. I was using the wrong character to make the second group non-capturing in the regex

**Comment by squirrelsc** [comment]
@LiliDeng LGTM

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**